presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -155,13 +155,19 @@ private boolean hasPreviousSpillCompletedSuccessfully()
     public ListenableFuture<?> startMemoryRevoke()
     {
         checkState(spillInProgress.isDone());
+        if (hashAggregationBuilder.hasBuiltFinalResult()) {
+            return spillInProgress;
+        }
```

Comment:
If I understand correctly, the original logic can only happen if an operator that has never spilled and just started to build final results; however, a revoke request comes in. Am I right? If that is the case, shall we make a comment here?

Reply:
Yeah so essentially we are declining memory revoking in the case that the hashaggbuilder has already completed. This is only set to true when `InMemoryHashAggregationBuilder.buildResult` is called - NOT `InMemoryHashAggregationBuilder.buildHashSortedResult`. This is because after buildResult, spilling should be impossible, because it can no longer process any more input anyway.